x-pack/plugins/maps/public/shared/layers/sources/wms_source/wms_source.js

```diff
@@ -83,9 +84,36 @@ export class WMSSource extends AbstractTMSSource {
     return this._descriptor.serviceUrl;
   }
 
+  /**
+   * Extend any query parameters supplied in the URL but override with required defaults
+   * (ex. srs must be EPSG:3857 to display in the map).
+   */
   getUrlTemplate() {
     const styles = this._descriptor.styles || '';
-    // eslint-disable-next-line max-len
-    return `${this._descriptor.serviceUrl}?bbox={bbox-epsg-3857}&format=image/png&service=WMS&version=1.1.1&request=GetMap&srs=EPSG:3857&transparent=true&width=256&height=256&layers=${this._descriptor.layers}&styles=${styles}`;
+    const layerUrl = parse(this._descriptor.serviceUrl, true);
```

Comment:
How about moving the getUrlTemplate generation logic into WmsClient?  Then generating the urlTemplate and the getCapabilities URL can share common functionality.

```
  _createUrl(defaultQueryParams) {
    const serviceUrl = parse(this._serviceUrl, true);
    const queryParams = {
      ...serviceUrl.query,
      ...defaultQueryParams,
    };
    return format({
      protocol: serviceUrl.protocol,
      hostname: serviceUrl.hostname,
      port: serviceUrl.port,
      pathname: serviceUrl.pathname,
      query: queryParams
    });
  }

  getUrlTemplate(layers, styles) {
    const urlTemplate = this._createUrl({
        format: 'image/png',
        service: 'WMS',
        version: '1.1.1',
        request: 'GetMap',
        srs: 'EPSG:3857',
        transparent: 'true',
        width: '256',
        height: '256',
        layers,
        styles
      });
    //TODO Find a better way to avoid URL encoding the template braces
    return `${urlTemplate}&bbox={bbox-epsg-3857}`
  }

  /**
   * Extend any query parameters supplied in the URL but override with required defaults
   * (ex. service must be WMS)
   */
  async _fetchCapabilities() {
    const getCapabilitiesUrl = this._createUrl({
        version: '1.1.1',
        request: 'GetCapabilities',
        service: 'WMS'
      });
    const resp = await this._fetch(getCapabilitiesUrl);
    if (resp.status >= 400) {
      throw new Error(`Unable to access ${this.state.serviceUrl}`);
    }
    const body = await resp.text();

    const parsePromise = new Promise((resolve, reject) => {
      parseString(body, (error, result) => {
        if (error) {
          reject(error);
        } else {
          resolve(result);
        }
      });
    });
    return await parsePromise;
  }
```